src/writable.ts

```diff
@@ -36,6 +35,17 @@ class WritableBundle<T> {
 		resources: Resource[],
 		signOptions?: SignOptions,
 	) {
+		const resourceIds = resources.map(({ id }) => id);
+		const duplicateIds = resourceIds.filter(
+			(resource, index) => resourceIds.indexOf(resource) !== index,
+		);
+		const duplicateIdsFiltered = Array.from(new Set(duplicateIds));
+		if (duplicateIdsFiltered.length !== 0) {
```

Comment:
I don't fully understand what you're doing here. Presumably, the following should be enough to determine if there are duplicate IDs in `resources`:

```suggestion
		const resourceIds = resources.map(({ id }) => id);
		const uniqueIds = new Set(resourceIds);
		if (resourceIds.length !== uniqueIds.size) {
```

If you want to communicate the duplicates, then within the `if` branch you could do:

```typescript
const duplicateIds = resourceIds.filter((id) => !uniqueIds.delete(id));
```

This takes advantage of `Set.delete` returning true/false depending on whether the item was in the set.

Reply:
Thanks, that is much better!